apps/src/aichat/views/ChatEventsList.tsx

```diff
@@ -21,36 +23,98 @@ const ChatEventsList: React.FunctionComponent<ChatEventsListProps> = ({
   const eventsString = JSON.stringify(events);
   const conversationContainerRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
+  const scrollToBottom = () => {
     if (conversationContainerRef.current) {
-      conversationContainerRef.current.scrollTo({
-        top: conversationContainerRef.current.scrollHeight,
-        behavior: 'smooth',
-      });
+      setShowScrollToBottom(false);
+
+      // On clear chat, we don't want to scroll to the bottom.
+      if (eventsString !== '[]') {
+        setInProgrammaticScroll(true);
+        conversationContainerRef.current.scrollTo({
+          top: conversationContainerRef.current.scrollHeight,
+          behavior: 'smooth',
+        });
+
+        const intervalId = setInterval(() => {
+          if (isAtBottom()) {
+            setInProgrammaticScroll(false);
+            clearInterval(intervalId);
+          }
+        }, 100);
+      }
+    }
+  };
+
+  const isAtBottom = () => {
+    const container = conversationContainerRef.current;
+
+    if (!container) {
+      return false;
     }
-  }, [eventsString, isWaitingForChatResponse]);
+
+    // Add a pixel of buffer to account for rounding errors.
+    return (
+      container.scrollTop + container.clientHeight + 1 >= container.scrollHeight
+    );
+  };
+
+  useEffect(() => {
+    const container = conversationContainerRef.current;
+
+    const handleScroll = () => {
+      if (container && !inProgrammaticScroll) {
+        setShowScrollToBottom(!isAtBottom());
+      }
+    };
+
+    container?.addEventListener('scroll', handleScroll);
+
+    return () => {
+      container?.removeEventListener('scroll', handleScroll);
+    };
+  }, [inProgrammaticScroll]);
+
+  useEffect(scrollToBottom, [eventsString, isWaitingForChatResponse]);
 
   return (
     <div
       id="chat-workspace-conversation"
-      className={moduleStyles.conversationArea}
-      ref={conversationContainerRef}
+      className={classNames(
+        moduleStyles.conversationArea,
+        moduleStyles.scrollToBottomContainer
+      )}
     >
-      {events.map(event => (
-        <ChatEventView
-          event={event}
-          key={event.timestamp}
-          isTeacherView={isTeacherView}
-        />
-      ))}
-      <WaitingAnimation shouldDisplay={isWaitingForChatResponse} />
+      <div className={moduleStyles.messageArea} ref={conversationContainerRef}>
+        {events.map(event => (
+          <ChatEventView
+            event={event}
+            key={event.timestamp}
+            isTeacherView={isTeacherView}
+          />
+        ))}
+        <WaitingAnimation shouldDisplay={isWaitingForChatResponse} />
+      </div>
+      {showScrollToBottom && (
+        <div className={moduleStyles.floatingScrollToBottomButtonContainer}>
+          <Button
+            isIconOnly
+            icon={{iconName: 'arrow-down'}}
+            size="s"
+            color="black"
+            type="secondary"
+            onClick={scrollToBottom}
```

Comment:
Based on [this Slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1737039205604889?thread_ts=1736870409.018249&cid=C06FELVTV0Q) could you add styling so button is circular?

Reply:
Updated!